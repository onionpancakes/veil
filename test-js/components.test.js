import React from 'react';
import renderer from 'react-test-renderer';

import "./components";

const {
    Widget,
    WidgetNested,
    WidgetKeyword,
    WidgetMapProps,
    WidgetFor,
    WidgetFragment
} = dev.onionpancakes.veil.test_js.components;


it("Widget", () => {
    let component = renderer.create(
        <Widget></Widget>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("WidgetNested", () => {
    let component = renderer.create(
        <WidgetNested></WidgetNested>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("WidgetKeyword", () => {
    let component = renderer.create(
        <WidgetKeyword></WidgetKeyword>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("WidgetMapProps", () => {
    let component = renderer.create(
        <WidgetMapProps></WidgetMapProps>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("WidgetFor", () => {
    let component = renderer.create(
        <WidgetFor></WidgetFor>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});

it("WidgetFragment", () => {
    let component = renderer.create(
        <div>
            <WidgetFragment></WidgetFragment>
        </div>
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});
