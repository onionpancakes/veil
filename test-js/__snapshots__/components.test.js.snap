// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Widget 1`] = `
<div>
  text
</div>
`;

exports[`WidgetFor 1`] = `
<div>
  <p>
    text0
  </p>
  <p>
    text1
  </p>
  <p>
    text2
  </p>
  <p>
    text3
  </p>
  <p>
    text4
  </p>
</div>
`;

exports[`WidgetFragment 1`] = `
<div>
  <div>
    foo
  </div>
  <div>
    bar
  </div>
</div>
`;

exports[`WidgetKeyword 1`] = `
<div
  className="bar baz"
  id="foo"
>
  text
</div>
`;

exports[`WidgetMapProps 1`] = `
<div
  className="yes1 yes2"
  id="foo"
>
  text
</div>
`;

exports[`WidgetNested 1`] = `
<div>
  <div>
    text
  </div>
</div>
`;
